content/docs/iac/build-with-pulumi/local-packages.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Local Packages
+meta_desc: This page provides an overview of working with locally generated Pulumi packages.
+menu:
+  iac:
+    parent: iac-build-with-pulumi
+    weight: 65
+aliases:
+  - /docs/using-pulumi/pulumi-packages/local-packages/
+---
+
+Pulumi Packages are the core technology that enables cloud infrastructure resource provisioning to be defined once and made available to users in all Pulumi languages. While many packages are published to the [Pulumi Registry](https://www.pulumi.com/registry/), there are cases where you might need to work with locally generated packages that haven't been published.
+
+## What are Local Packages?
+
+Local packages are Pulumi packages with sdks generated on your computer, instead of being checked into a provider repository and published to the Pulumi Registry. All Pulumi packages include a [schema](/docs/iac/using-pulumi/pulumi-packages/schema/) that defines the resources and functions exposed by the package. These packages are frequently part of the Pulumi experience in several contexts:
+
+- When using parameterized providers like [pulumi-terraform-provider](https://www.pulumi.com/registry/packages/terraform-provider/).
+- When working with components as packages.
+- During package development and testing.
+- For private or organization-specific resources.
+
+Local packages allow you to generate SDKs in any Pulumi language for resources or components before they're published.
+
+## Adding Local Packages with `pulumi package add`
+
+The preferred way to add a local package is with the `pulumi package add` command. This command:
+
+1. Locally generates an SDK in your currently selected Pulumi language.
+1. Adds the package to your project configuration file (Pulumi.yaml).
+1. Prints instructions on how to link the package into your project.
+
+### Command Syntax
+
+```bash
+pulumi package add <provider|schema|path> [provider-parameter...] [flags]
+```
+
+### Adding a Package from Different Sources
+
+The `pulumi package add` command can accept the package source in multiple ways:
+
+#### From a Plugin Reference
```

Comment:
What does this do? I think we're missing accompanying docs for what a plugin actually is. (We might need to add some supporting pages before it's possible to publish this page with complete info.)